### DISCUSSION TITLE
Remove deleted/purged data from bucket not possible?

### DISCUSSION THREAD

**Discussion by pablitovicente** [nats-io/nats-server#6015] [Q&A] [answered]
I am using KV Store and sometimes I need to delete keys that might never exist again.

I am first deleting the keys and then also purging but the bucket never shrinks and the watcher gets the purge updates.

I tried the compact command but this still does not remove the purged keys.

Is there another command to get rid of purged values? My concern is what happens if I have 3 gigabytes of deleted/purged values I will never access or use those values again?

**Comment by derekcollison** [comment]
Do you have some sample code you could share?

**Comment by jnmoyne** [comment, marked as answer]
> Is there another command to get rid of purged values? My concern is what happens if I have 3 gigabytes of deleted/purged values I will never access or use those values again?

At this point your available workaround is to do a purge (specifying the key(s) you want to delete as part of the subject filter passed to that purge command) directly on the stream underlying your KV bucket after or instead of doing the KV delete for those key(s).

**Reply by pablitovicente** [reply]
Oh gotcha that could actually work well as my keys anyway have the format namespace.subnamespace.x.y.z I will try and report back

**Comment by pablitovicente** [comment]
I will resolve this one as purging the stream subject works. 

One detail is that the subjects are also prefixed for KV with '$KV' so in my case I had 'foo.bar.>' and had to pass as filter '$KV.foo.bar.>'.

Is there a reason why this is not exposed in the SDKs? It seems to be very useful for some use cases.
